src/OpenTelemetry.ResourceDetectors.Host/HostDetector.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using OpenTelemetry.Resources;
+
+namespace OpenTelemetry.ResourceDetectors.Host;
+
+/// <summary>
+/// Host detector.
+/// </summary>
+public sealed class HostDetector : IResourceDetector
+{
+    /// <summary>
+    /// Detects the resource attributes from host.
+    /// </summary>
+    /// <returns>Resource with key-value pairs of resource attributes.</returns>
+    public Resource Detect()
+    {
+        try
+        {
+            return new Resource(new List<KeyValuePair<string, object>>(1)
+            {
+                new(HostSemanticConventions.AttributeHostName, Environment.MachineName),
+            });
+        }
+        catch (InvalidOperationException)
+        {
+            // do nothing if there is not possibility to fetch host name
```

Comment:
please add a todo to log to usual logging mechanism.

Reply:
996e12b18ec6cba031186cb803f88b42239782b2